misk/src/main/kotlin/misk/scope/ActionScopedProvider.kt

```diff
@@ -0,0 +1,9 @@
+package misk.scope
+
+/**
+ * An [ActionScopedProvider] is implemented by components and application code that wants
+ * provide contextual information based on an incoming request, job data, etc.
+ */
+interface ActionScopedProvider<out T> {
```

Comment:
I think our names or abstractions aren’t simple enough yet. I’ll have a tough time explaining the difference between this and ActionScoped!

Reply:
yeah i didn't have a good name for this - opinions welcome